tools/raisinbread_refresh.php

```diff
@@ -0,0 +1,286 @@
+    }
+    echo PHP_EOL . 'Please enter the name of your database:' . PHP_EOL;
+    $dbname = trim(fgets(STDIN));
+} catch (\LorisException $e) {
+    die($e->getMessage());
+} catch (Exception $e) {
+    die("Could not load project/config.xml");
+}
+
+
+echo <<<CONFIRMATION
+Please type the database name `$dbname` to confirm you wish to drop tables
+and import test data: 
+CONFIRMATION;
+
+$input = trim(fgets(STDIN));
+if ($input !== $dbname) {
+    die('Input did not match database name. Exiting.');
+}
+
+$mysqlCommand = <<<CMD
+mysql -A $dbname
+CMD;
+
+// Test whether a connection to MySQL is possible via a config file.
+// If not, read DB information from the config file. This method is not as 
+// preferable because it generates MySQL warnings due to the password being
+// supplied via a command-line argument.
+exec($mysqlCommand . ' -e "show tables;" 2>&1 1>/dev/null', $output, $status);
+if ($status != 0) {
+    $mysqlCommand = <<<CMD
+mysql -A "$dbname" -u "$username" -h "$host" -p$password
+CMD;
+}
+
+// Drop tables
+echo PHP_EOL .'Dropping LORIS tables....' . PHP_EOL;
+dropTables();
+
+// Print the names of remaining tables, if any. Some tables may remain if they
+// have been created at a different time during development and are not
+// deleted by the above script. Issues with e.g. foreign keys can be a source
+// of error and so it is useful to know what tables remain.
+exec(
+    "$mysqlCommand -e 'SHOW TABLES;'",
+    $tables,
+    $status
+);
+array_shift($tables); // remove the column name
+if (count($tables) > 0) {
+    echo "\e[33mWARNING: Untracked tables still exist in the database:\e[0m\n";
+    array_walk($tables, 'printBulletPoint');
+
+    echo "Do you want to delete them now? (y/N)" . PHP_EOL;
+    $input = trim(fgets(STDIN));
+    if (mb_strtolower($input) === 'y') {
+        dropRemainingTables($tables);
+    }
+}
+
+// Source LORIS core tabes
+echo <<<INFO
+[*] Creating LORIS core tables.... 
+
+INFO;
+$coreTables = glob("../SQL/0000-*.sql"); 
+array_walk($coreTables, 'runPatch');
+
+// Create instrument tables
+echo <<<INFO
+[*] Creating instrument tables.... 
+
+INFO;
+$rbInstrumentTables = glob("../raisinbread/instruments/instrument_sql/*.sql"); 
+array_walk($rbInstrumentTables, 'runPatch');
+
+// Import Raisinbread data
+echo <<<INFO
+[*] Importing Raisinbread data....
+
+INFO;
+$rbData= glob("../raisinbread/RB_files/*.sql");
+array_walk($rbData, 'runPatch');
+
+// Restore config settings if they were successfully found before.
+if (isset($urlConfigSetting)) {
+    restoreConfigSetting('url', $urlConfigSetting);
+}
+if (isset($baseConfigSetting)) {
+    restoreConfigSetting('base', $baseConfigSetting);
+}
+if (isset($hostConfigSetting)) {
+    restoreConfigSetting('host', $hostConfigSetting);
+}
+
+// Trigger a password reset because the password for `admin` in the Raisinbread
+// database is public.
+echo 'Please choose a new password for the admin user:' . PHP_EOL;
+runCommand('php resetpassword.php admin');
```

Comment:
There's a prompt on line 175. I don't understand why we need another one

Reply:
<img width="342" alt="Screen Shot 2020-01-15 at 14 24 36" src="https://user-images.githubusercontent.com/4022790/72464374-20efba00-37a3-11ea-8633-9caf1383b606.png">

Reply:
because the `Running command:...` displays after the `Please choose...`

I waited a good 5 min before realising that i had to enter the password now.

If these 2 lines can be inversed or the `Please choose ...` can be removed entirely and have the resetpassword script promt itself for the new password would make more sens

Reply:
OK I'll change it